internal/stringutil/uuid.go

```diff
@@ -3,19 +3,16 @@ package stringutil
 import (
 	"fmt"
 	"math/rand"
-	"sync"
-	"time"
 )
 
-var uuidRand = rand.New(rand.NewSource(time.Now().UnixNano()))
-var uuidMutex = &sync.Mutex{}
-
-// UUID generates a random UUID according to RFC 4122.
-func UUID() string {
+// UUID generates a random UUID according to RFC 4122, using optional rand if supplied
+func UUID(rs rand.Source) string {
+	rand.Float64()
```

Comment:
this is actually how the controlled randomness gets passed into uuid func.

Reply:
Sorry, it appears GitHub doesn't highlight the exact line selected for the comment.  I was referring to the `rand.Float64()` line, which isn't using the value returned.   If you are intentionally priming the random generation or something, please add a comment to that effect, since it's purpose is unclear.

Reply:
ah yes, the errant rand.Float64(), yes. put that in for some debugging and then forgot to remove.